src/conversions/bytes.rs

```diff
@@ -0,0 +1,161 @@
+#![cfg(feature = "bytes")]
+
+//! Conversions to and from [bytes](https://docs.rs/bytes/latest/bytes/)'s [`Bytes`] and
+//! [`BytesMut`] types.
+//!
+//! This is useful for efficiently converting Python's `bytes` and `bytearray` types efficiently.
+//!
+//! # Setup
+//!
+//! To use this feature, add in your **`Cargo.toml`**:
+//!
+//! ```toml
+//! [dependencies]
+#![doc = concat!("pyo3 = { version = \"", env!("CARGO_PKG_VERSION"),  "\", features = [\"bytes\"] }")]
+//! bytes = "1.10"
+//!
+//! Note that you must use compatible versions of bytes and PyO3.
+//!
+//! # Example
+//!
+//! Rust code to create functions which return `Bytes` or take `Bytes` as arguements:
+//!
+//! ```rust,no_run
+//! use pyo3::prelude::*;
+//!
+//! #[pyfunction]
+//! fn get_message_bytes() -> Bytes {
+//!     Bytes::from(b"Hello Python!".to_vec())
+//! }
+//!
+//! #[pyfunction]
+//! fn num_bytes(bytes: Bytes) -> usize {
+//!     bytes.len()
+//! }
```

Comment:
The example is not that great, the user should just use `&[u8]` in this case. Please change the example to something where the Bytes is actually needed.

Reply:
... that said, we also need to be careful to steer users away from #2888, unless we can fix it.